### PR TITLE
display extra info at the bottom of invoice 

### PR DESCRIPTION
**Pull request by vinuch**
resolve opencollective/opencollective#2907

added a paragraph to display the content of invoice.extraInfo if any

### CODE REVIEW

File: src/pages/invoice.js [hunk 1]
```diff
@@ -321,7 +323,7 @@ export class InvoicePage extends React.Component {
 
                   <Box>
                     <H2>
-                      {invoice.title || (
+                      {invoice.title|| (
```
Comment (Betree) [inline src/pages/invoice.js]:
```suggestion
                      {invoice.title || (
```

File: src/pages/invoice.js [hunk 2]
```diff
@@ -371,18 +373,29 @@ export class InvoicePage extends React.Component {
...
                         fontWeight="bold"
                       >
                         <FormattedMessage id="total" defaultMessage="TOTAL" />
                         <Span>{formatCurrency(invoice.totalAmount, invoice.currency)}</Span>
+                        
                       </Container>
+                      
                     </Container>
+
+
```
Comment (Betree) [inline src/pages/invoice.js]:
```suggestion   
```

File: src/pages/invoice.js [hunk 3]
```diff
@@ -371,18 +373,29 @@ export class InvoicePage extends React.Component {
...
                         px={3}
                         py={2}
                         background="#ebf4ff"
                         fontWeight="bold"
                       >
                         <FormattedMessage id="total" defaultMessage="TOTAL" />
                         <Span>{formatCurrency(invoice.totalAmount, invoice.currency)}</Span>
+                        
                       </Container>
+
```
Comment (Betree) [inline src/pages/invoice.js]:
```suggestion
```

File: src/pages/invoice.js [hunk 4]
```diff
@@ -371,18 +373,29 @@ export class InvoicePage extends React.Component {
...
                         justifyContent="space-between"
+                        flexBasis="100%"
                         px={3}
                         py={2}
                         background="#ebf4ff"
                         fontWeight="bold"
                       >
                         <FormattedMessage id="total" defaultMessage="TOTAL" />
                         <Span>{formatCurrency(invoice.totalAmount, invoice.currency)}</Span>
+
```
Comment (Betree) [inline src/pages/invoice.js]:
```suggestion
```

File: src/pages/invoice.js [hunk 5]
```diff
@@ -371,18 +373,29 @@ export class InvoicePage extends React.Component {
...
+                        
                       </Container>
+                      
                     </Container>
+
+                    
                   </Flex>
                 )}
+                
+
```
Comment (Betree) [inline src/pages/invoice.js]:
```suggestion
```

File: src/pages/invoice.js [hunk 6]
```diff
@@ -371,18 +373,29 @@ export class InvoicePage extends React.Component {
...
+                    
                   </Flex>
                 )}
+                
+               
               </Box>
+              <P mt={2} textAlign="left" color="black" padding="4rem 0">
+                  {invoice.extraInfo || 
+                  <FormattedMessage id="invoice.donationAppreciation" defaultMessage="" />}
+                </P>
```
Comment (Betree) [inline src/pages/invoice.js]: The linter should not accept that, code is not formatted properly. Can you run `npm run prettier` to fix it?

File: src/pages/invoice.js [hunk 7]
```diff
@@ -371,18 +373,29 @@ export class InvoicePage extends React.Component {
...
+
+                    
                   </Flex>
                 )}
+                
+               
               </Box>
+              <P mt={2} textAlign="left" color="black" padding="4rem 0">
+                  {invoice.extraInfo || 
+                  <FormattedMessage id="invoice.donationAppreciation" defaultMessage="" />}
```
Comment (Betree) [inline src/pages/invoice.js]: I don't understand the purpose of this `FormattedMessage`?

Comment (vinuch) [reply src/pages/invoice.js]: i was your using it to display a message when it was not connected to the api ill take it out now

File: .env.local [hunk 8]
```diff
@@ -1,6 +1,6 @@
 PORT=3002
 NODE_ENV=development
 WEBSITE_URL=http://localhost:3000
-API_URL=http://localhost:3060
-API_KEY=dvl-1510egmf4a23d80342403fb599qd
+API_URL=http://localhost:3002
+API_KEY=dd83d087d9efddf31c11c31282c5383fc08768253
 LOG_LEVEL=debug
```
[the remark this answers is not in the document]
Comment (vinuch) [reply .env.local]: returned this back to how it was

File: src/pages/invoice.js (hunk 6 above)
Comment (vinuch) [reply src/pages/invoice.js]: Okay i have done this did that

File: package-lock.json [hunk 9]
```diff
@@ -6134,24 +6134,21 @@
       "version": "2.0.0",
```
[the remark this answers is not in the document]
Comment (vinuch) [reply package-lock.json]: ohh okay do i add it to .gitignore?

[the remark this answers is not in the document]
Comment (Betree) [reply package-lock.json]: @vinuch please don't mark the conversation as resolved before the issue is properly addressed, the file is still there.

No need to add it to .gitignore, we want to keep this file in sync. Only there's no reason to commit it if there's no changes in dependencies.

See https://docs.opencollective.com/help/contributing/development/best-practice-guidelines

File: src/pages/invoice.js [hunk 10]
```diff
@@ -383,6 +384,9 @@ export class InvoicePage extends React.Component {
                   </Flex>
                 )}
               </Box>
+              <P mt={2} textAlign="left" color="black" padding="4rem 0">
+                {invoice.extraInfo}
+              </P>
```
Comment (Betree) [inline src/pages/invoice.js]: Extra info may not be there. Let's be safe:

```suggestion
              {invoice.extraInfo && (
                <P mt={2} textAlign="left" color="black" padding="4rem 0">
                  {invoice.extraInfo}
                </P>
              )}
```

File: package-lock.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (Betree) [reply package-lock.json]: @vinuch You again marked this as resolved despite the fact that `package-lock.json` is still there. Please don't do that.

[the remark this answers is not in the document]
Comment (vinuch) [reply package-lock.json]: @Betree  should i delete the `package-lock.json` and push? would that do it?

**Comment by joshikeno** [comment]
@vinuch please, resolve the conflicts.

**Comment by vinuch** [comment]
@joshikeno I can’t see what the conflicts are could you point me to them?

**Comment by vinuch** [comment]
hey please im confused i dont know what to do now i see there re conflicts but when i pull it doesnt show up on my local ....i could you use some help ...thanks.
@Betree @joshikeno

**Comment by joshikeno** [comment]
@vinuch  You need to update your branch to sync with master. ‘git pull —rebase —autostash origin master’

**Comment by vinuch** [comment]
@joshikeno  running that it says my current branch is up to date

**Comment by Betree** [comment]
@vinuch See https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/merging-an-upstream-repository-into-your-fork. Please note that the repository has been renamed to `opencollective/opencollective-pdf`.

Feel free to open a new PR if solving the conflicts is too complex, a lot has changed in this project in two months.

**Comment by vinuch** [comment]
@Betree  the link was helpful i was able to resolve the conflicts and the checks have passed. i think its ready for review

File: package-lock.json [hunk 11]
```diff
@@ -6744,9 +6744,9 @@
       "integrity": "sha512-GpVkmM8vF2vQUkj2LvZmD35JxeJOLCwJ9cUkugyk2nuhbv3+mJvpLYYt+0+USMxE+oj+ey/lJEnhZw75x/OMcQ=="
     },
     "commitizen": {
-      "version": "4.0.3",
```
Comment (Betree) [inline package-lock.json:6747]: This file should not have changes

File: package.json [hunk 12]
```diff
@@ -28,6 +28,7 @@
     "express-winston": "~4.0.3",
     "graphql": "14.6.0",
     "graphql-request": "~1.8.2",
+    "graphql-tag": "^2.10.3",
```
Comment (Betree) [inline package.json:31]: This file should not have changes

File: src/server/lib/graphql.js [hunk 13]
```diff
@@ -0,0 +1,154 @@
+import { GraphQLClient } from 'graphql-request';
```
Comment (Betree) [inline src/server/lib/graphql.js:1]: This file should not be commit. If you're looking for adding the `extraInfo` field on the invoice, I suggest to do it in https://github.com/opencollective/opencollective-invoices/blob/d53b048a97727c3f014ca671eab55fe1cfaa7b09/lib/graphql/fragments.js#L3.

File: package-lock.json (hunk 11 above)
Comment (Betree) [reply package-lock.json:6747]: Still there

**Comment by Betree** [comment]
Alright, good to go.

**Comment by Betree** [comment]
Followed up with a fix an enhancements in https://github.com/opencollective/opencollective-pdf/pull/348

**Pull request merged**